app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,61 @@
+class ReviewsController < ApplicationController
+  def new
+    @restaurant= Restaurant.find(params[:restaurant_id])
+    @review = Review.new
+  end
+
+  def create
+    @restaurant= Restaurant.find(params[:restaurant_id])
+    @review = @restaurant.reviews.new(review_params)
+    @review.user = current_user
+    if @review.save
+      flash[:notice] = "Review Created!"
+      redirect_to restaurant_path(@restaurant)
+    else
+      render :new
+    end
+  end
+
+    def destroy
+      @restaurant = Restaurant.find(params[:restaurant_id])
+      @review = Review.find(params[:id])
+
+      if @review.user_id != current_user.id
+        flash[:notice] = "You cannot delete another user's review!"
+        render :"restaurants/show"
+      else
+        @review.destroy
+        flash[:notice] = "Review Deleted!"
+        redirect_to restaurant_path(@restaurant)
+      end
+    end
+
+    def edit
+      @restaurant = Restaurant.find(params[:restaurant_id])
+      @review = Review.find(params[:id])
+      # render plain: params[:review].inspect
+    end
+
+    def update
```

Comment:
Assignment Branch Condition size for update is too high. [17.23/15]<br>Inconsistent indentation detected.